packages/core/database/lib/entity-manager/regular-relations.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const {
+  isBidirectional,
+  isOneToAny,
+  isManyToAny,
+  isAnyToOne,
+  isAnyToMany,
+} = require('../metadata/relations');
+const { createQueryBuilder } = require('../query');
+
+const getSelect = (joinTable, attribute) => {
+  const { joinColumn, orderColumnName, inverseJoinColumn, inverseOrderColumnName } = joinTable;
+  const select = [joinColumn.name, inverseJoinColumn.name];
+  if (isAnyToMany(attribute)) {
+    select.push(orderColumnName);
+  }
+  if (isBidirectional(attribute) && isManyToAny(attribute)) {
+    select.push(inverseOrderColumnName);
+  }
+  return select;
+};
+
+const deletePreviousOneToAnyRelations = async ({ id, attribute, joinTable, relIdsToadd, db }) => {
+  const { joinColumn, inverseJoinColumn, orderColumnName } = joinTable;
+  const select = getSelect(joinTable, attribute);
+
+  // need to delete the previous relations for oneToAny relations
+  if (isBidirectional(attribute) && isOneToAny(attribute)) {
+    // update orders for previous oneToAny relations that will be deleted if it has order (oneToMany)
+    if (isAnyToMany(attribute)) {
+      const currentRelsToDelete = await createQueryBuilder(joinTable.name, db)
+        .select(select)
+        .where({
+          [inverseJoinColumn.name]: relIdsToadd,
+          [joinColumn.name]: { $ne: id },
+        })
+        .where(joinTable.on || {})
+        .execute();
+
+      currentRelsToDelete.sort((a, b) => b[orderColumnName] - a[orderColumnName]);
+
+      for (const relToDelete of currentRelsToDelete) {
+        if (relToDelete[orderColumnName] !== null) {
+          await createQueryBuilder(joinTable.name, db)
+            .decrement(orderColumnName, 1)
+            .where({
+              [joinColumn.name]: relToDelete[joinColumn.name],
+              [orderColumnName]: { $gt: relToDelete[orderColumnName] },
+            })
+            .where(joinTable.on || {})
+            .execute();
+        }
+      }
+    }
```

Comment:
Maybe it makes this too complicated to read, but I think this can be done with a single query.
https://github.com/strapi/experiments/pull/1/files#diff-c253cf0e2025afa2af251a04e4dcbe8deeec721161e95ef719b758bc2cc78e90R24

Didn't spend time doing it with knexjs query builder but I can try if you find it addecuate.

Reply:
Wow, thank you very much, I'll use that! 😀

Reply:
Wow, row_number rocks. Forgot about that one.
Makes complete sense. You just reorder everything from the start with a single query, no decrements, no math.

Reply:
Actually, you do not need the partition by, only with:
```
SELECT id, ROW_NUMBER() OVER (ORDER BY order) AS rowNum FROM tes_table
```

Reply:
I think we need the partition here 🤔 it won't give the right order per document but a global order for all the relations of all documents otherwise.

Reply:
True, forgot we had two order columns